cupy/elementwise.py

```diff
@@ -501,19 +508,43 @@ def _guess_routine_from_dtype(ops, dtype):
 
 def _guess_routine(name, cache, ops, in_args, dtype):
     if dtype is None:
-        key = tuple([numpy.dtype(type(i)).type
-                     if isinstance(i, _python_scalar_type) else i.dtype.type
-                     for i in in_args])
-    else:
-        key = dtype
-
-    op = cache.get(key, ())
-    if op is ():
-        if dtype is None:
-            op = _guess_routine_from_in_types(ops, key)
+        all_scalars = True
```

Comment:
Please extract the routine for determining the value of `use_min_scalar` as a separate function.